src/VisualStudio/Core/Def/Implementation/ChangeSignature/ChangeSignatureDialogViewModel.cs

```diff
@@ -440,89 +440,117 @@ public string RestoreAutomationText
-                get { return $"{Type} {Parameter}"; }
-            }
+            public string ParameterAutomationText => $"{Type} {Parameter}";
 
             public string Modifier
             {
                 get
                 {
-                    // Todo: support VB
-                    switch (_parameter.RefKind)
+                    string CS()
                     {
-                        case RefKind.Out:
-                            return "out";
-                        case RefKind.Ref:
-                            return "ref";
-                        case RefKind.In:
-                            return "in";
+                        switch (ParameterSymbol.RefKind)
+                        {
+                            case RefKind.Out:
+                                return "out";
+                            case RefKind.Ref:
+                                return "ref";
+                            case RefKind.In:
+                                return "in";
+                        }
+
+                        if (ParameterSymbol.IsParams)
+                        {
+                            return "params";
+                        }
+
+                        if (_changeSignatureDialogViewModel._thisParameter != null &&
+                            ParameterSymbol == _changeSignatureDialogViewModel._thisParameter.ParameterSymbol)
+                        {
+                            return "this";
+                        }
+                        return string.Empty;
                     }
 
-                    if (_parameter.IsParams)
+                    string VB()
                     {
-                        return "params";
+                        switch (ParameterSymbol.RefKind)
+                        {
+                            case RefKind.Ref:
+                                return "ByRef";
+                        }
+
+                        if (ParameterSymbol.IsParams)
+                        {
+                            return "ParamArray";
+                        }
+
+                        if (_changeSignatureDialogViewModel._thisParameter != null &&
+                            ParameterSymbol == _changeSignatureDialogViewModel._thisParameter.ParameterSymbol)
+                        {
+                            return "Me";
+                        }
+                        return string.Empty;
                     }
 
-                    if (_changeSignatureDialogViewModel._thisParameter != null &&
-                        _parameter == _changeSignatureDialogViewModel._thisParameter._parameter)
+                    switch (ParameterSymbol.Language)
                     {
-                        return "this";
+                        case "C#":
+                            return CS();
+                        case "Visual Basic":
+                            return VB();
+                        default:
+                            return string.Empty;
```

Comment:
i don't see any benefit to these local function which you then immediately invoke once.  why not just test the language and have the language specific code in the if-block?